src/main/java/org/apache/sling/xss/impl/xml/PolicyProvider.java

```diff
@@ -0,0 +1,142 @@
+            this.cssRules = Collections.unmodifiableMap(cssrules);
+            this.elementMatcher = new IncludeExcludeMatcher(commonRegExps.get("cssElementSelector"),
+                    commonRegExps.get("cssElementExclusion"));
+            this.classMatcher = new IncludeExcludeMatcher(commonRegExps.get("cssClassSelector"),
+                    commonRegExps.get("cssClassExclusion"));
+            this.idMatcher = new IncludeExcludeMatcher(commonRegExps.get("cssIDSelector"),
+                    commonRegExps.get("cssIDExclusion"));
+            this.pseudoElementMatcher = new IncludeExcludeMatcher(commonRegExps.get("cssPseudoElementSelector"),
+                    commonRegExps.get("cssPseudoElementExclusion"));
+            this.attributeMatcher = new IncludeExcludeMatcher(commonRegExps.get("cssAttributeSelector"),
+                    commonRegExps.get("cssAttributeExclusion"));
+        }
+
+        public Map<String, Property> getCssRules() {
+            return cssRules;
+        }
+
+        public boolean isValidElementName(String name) {
+            return elementMatcher.matches(name);
+        }
+
+        public boolean isValidClassName(String name) {
+            return classMatcher.matches(name);
+        }
+
+        public boolean isValidId(String name) {
+            return idMatcher.matches(name);
+        }
+
+        public boolean isValidPseudoElementName(String name) {
+            return pseudoElementMatcher.matches(name);
+        }
+
+        public boolean isValidAttributeSelector(String name) {
+            return attributeMatcher.matches(name);
+        }
+    }
+
+    public Map<String, String> getDirectives() {
+        return directives;
+    }
+
+    public List<String> getRequireClosingTags() {
+        return requireClosingTags;
+    }
+
+    public Map<String, Pattern> getCommonRegularExpressions() {
+        return commonRegularExpressions;
+    }
+
+    public Map<String, Attribute> getGlobalAttributes() {
+        return globalAttributes;
+    }
+
+    public Map<String, Attribute> getCommonAttributes() {
+        return commonAttributes;
+    }
+
+    public Map<String, Property> getCssRules() {
+        return cssRules;
+    }
+
+    public List<String> getAllowedEmptyTags() {
+        return allowedEmptyTags;
+    }
+
+    public Map<String, Tag> getTagRules() {
+        return tagRules;
+    }
+
+    public Map<String, Attribute> getDynamicAttributes() {
+        return dynamicAttributes;
+    }
+
+    public CssPolicy getCssPolicy() {
+        return new CssPolicy(cssRules,
+                commonRegularExpressions, directives);
+    }
+
+    public PolicyProvider(InputStream input) throws PolicyException, XMLStreamException, IOException {
+        AntiSamyXmlParser xmlParser = new AntiSamyXmlParser();
+        MapBuilder mapBuilder = new MapBuilder();
+        AntiSamyRules root = null;
```

Comment:
Nit: collapse this line and the next one into a single one: 

```java
AntiSamyRules root = xmlParser.createRules(input);
```

Reply:
Resolved: 99bb91ab6e2529fd6ef25858f874a26cb159e87e